tests/user.js

```diff
@@ -0,0 +1,193 @@
+var app = require('../index');
+var request = require('supertest')(app);
+var expect = require('chai').expect;
+var User = require('../server/models/user');
+
+describe('user test suite', function () {
+  var token;
+  var userId;
+
+  before(function (done) {
+    request
+      .post('/api/users/login')
+      .send({
+        email: 'jwarugu@gmail.com',
+        password: 'jwarugu'
+      })
+      .end(function (err, res) {
+        if (err) { return done(err); }
+        token = res.body.token;
+        done();
+      });
+  });
+
+  describe('/users CRUD operations', function () {
+    it('route that returns all users', function (done) {
+      request
+        .get('/api/users')
+        .set('x-access-token', token)
+        .expect(200)
+        .end(function (err, res) {
+          expect(res.body).to.have.length(4);
+          expect(Array.isArray(res.body)).to.equal(true);
+          userId = res.body[0]._id;
+          done();
+        });
+    });
+
+    it('creates a user in the db', function (done) {
+      request
+        .post('/api/users')
+        .set('x-access-token', token)
+        .send({
+          username: 'user',
+          first: 'first',
+          last: 'last',
+          email: 'user@gmail.com',
+          password: 'user',
+          role: 'user'
+        })
+        .expect(200)
+        .expect({message: 'New user created'}, done);
+    });
+
+    it('asserts that no duplicates can be created', function (done) {
+      var username = User.schema.paths.username;
+      var email = User.schema.paths.email;
+
+      expect(username.options.index.unique).to.equal(true);
+      expect(email.options.index.unique).to.equal(true);
+
+      request
+        .post('/api/users')
+        .set('x-access-token', token)
+        .send({
+          username: 'user',
+          first: 'first',
+          last: 'last',
+          email: 'user@gmail.com',
+          password: 'user',
+          role: 'user'
+        })
+        .expect(409)
+        .expect({message: 'Duplicate entry'}, done);
+    });
+
+    it('asserts that last and first names are required', function (done) {
+      request
+        .post('/api/users')
+        .set('x-access-token', token)
+        .send({
+          username: 'test',
+          email: 'test@gmail.com',
+          password: 'test',
+          role: 'user'
+        })
+        .expect(400)
+        .expect({message: 'Error occured while saving the user'}, done);
+    });
+
+    it('asserts that a new user has a role defined', function (done) {
+      var role = User.schema.paths.role;
+      expect(role.options.required).to.equal(true);
```

Comment:
Ideally, consider returning the user created so that you can confirm the user has a role defined. Minus sensitive details of course. 👍